be/src/io/cache/cache_lru_dumper.cpp

```diff
@@ -0,0 +1,375 @@
+                fmt::format("dump lru reading failed, file={}, {}", filename, err_msg.str()));
+        LOG(WARNING) << warn_msg;
+        return Status::InternalError<false>(warn_msg);
+    }
+
+    return Status::OK();
+}
+
+Status CacheLRUDumper::dump_one_lru_entry(std::ofstream& out, std::string& filename,
+                                          const UInt128Wrapper& hash, size_t offset, size_t size) {
+    // Dump file format description:
+    // +-----------------------------------------------+
+    // | LRUDumpEntryGroupPb_1                         |
+    // +-----------------------------------------------+
+    // | LRUDumpEntryGroupPb_2                         |
+    // +-----------------------------------------------+
+    // | LRUDumpEntryGroupPb_3                         |
+    // +-----------------------------------------------+
+    // | ...                                           |
+    // +-----------------------------------------------+
+    // | LRUDumpEntryGroupPb_n                         |
+    // +-----------------------------------------------+
+    // | LRUDumpMetaPb (queueName, Group<off,size>List)|
+    // +-----------------------------------------------+
+    // | FOOTER_OFFSET (8Bytes)                        |
+    // +-----------------------------------------------+
+    // | CHECKSUM (4Bytes)｜VERSION (1Byte)｜ MAGIC     |
+    // +-----------------------------------------------+
+    //
+    // why we are not using protobuf as a whole?
+    // AFAIK, current protobuf version dose not support streaming mode,
+    // so that we need to store all the message in memory which will
+    // consume loads of RAMs.
+    // Instead, we use protobuf serialize each of the single entry
+    // and provide the version field in the footer for upgrade
+
+    ::doris::io::cache::LRUDumpEntryPb* entry = _current_dump_group.add_entries();
+    ::doris::io::cache::UInt128WrapperPb* hash_pb = entry->mutable_hash();
+    hash_pb->set_high(hash.high());
+    hash_pb->set_low(hash.low());
+    entry->set_offset(offset);
+    entry->set_size(size);
+
+    _current_dump_group_count++;
+    if (_current_dump_group_count >= 10000) {
+        RETURN_IF_ERROR(flush_current_group(out, filename));
+    }
+    return Status::OK();
+}
+
+Status CacheLRUDumper::flush_current_group(std::ofstream& out, std::string& filename) {
+    if (_current_dump_group_count == 0) {
+        return Status::OK();
+    }
+
+    // Record current position as group start offset
+    size_t group_start = out.tellp();
+
+    // Serialize and write the group
+    std::string serialized;
+    LOG(INFO) << "Before serialization: " << _current_dump_group.DebugString();
+    if (!_current_dump_group.SerializeToString(&serialized)) {
+        return Status::InternalError<false>("Failed to serialize LRUDumpEntryGroupPb");
+    }
+    LOG(INFO) << "Serialized size: " << serialized.size();
+    out.write(serialized.data(), serialized.size());
```

Comment:
this write should not block `dump_one_entry`
use a simple double buffer stretegy to isolate flush and dump one entry
e.g.
1. lock
2. create a new dump group, set current dump group to flush group (if no flush is in progress)
3. set current dump group with the newly created dump group
4. unlock

flush with the flush group